src/zebra_puzzles/evaluation.py

```diff
@@ -223,30 +233,90 @@ def query_llm(prompt: str, model: str, response_format: Type[BaseModel]) -> Base
             temperature=0,
             seed=42,
             response_format=response_format,
+            max_completion_tokens=100_000,
+            reasoning_effort="medium",
         )
     except BadRequestError as e:
-        if "'temperature' is not supported" in str(e):
-            response = client.beta.chat.completions.parse(
-                messages=[{"role": "user", "content": prompt}],
-                model=model,
-                seed=42,
-                response_format=response_format,
-            )
-        else:
-            raise e
+        max_retries = 5
+        wait_time = 5
+        # gpt-4o-mini
+        if "'Unrecognized request argument supplied: reasoning_effort'" in str(e):
+            while retries := 0 < max_retries:
+                try:
+                    response = client.beta.chat.completions.parse(
+                        messages=[{"role": "user", "content": prompt}],
+                        model=model,
+                        temperature=0,
+                        seed=42,
+                        response_format=response_format,
+                        max_completion_tokens=16_384,
+                    )
+                except (InternalServerError,APIError,APIConnectionError,RateLimitError) as e:
+                    retries += 1
+                    print(f"\nRetrying {retries} time(s) due to:\n{e}")
```

Comment:
Nit: Generally logs are preferred over prints. It's not too important in this repo, but it's good practice in the future to never use prints and use logs instead, as that means that the users of the code can easily turn it off and on, and you can have different sorts of messages (debug, info, warning, error).